pkg/apis/submariner.io/v1/types.go

```diff
@@ -78,8 +78,13 @@ type EndpointSpec struct {
 	NATEnabled    bool              `json:"nat_enabled"`
 	Backend       string            `json:"backend"`
 	BackendConfig map[string]string `json:"backend_config,omitempty"`
+	Ports         map[string]string `json:"ports,omitempty"`
```

Comment:
As discussed on slack:

```suggestion
	NATDiscoveryPort         *int32 `json:"natDiscoveryPort,omitempty"`
```